src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Byte.cs

```diff
@@ -1309,85 +1311,206 @@ public static unsafe int LastIndexOfAny(ref byte searchSpace, byte value0, byte
-                while ((byte*)lengthToExamine > (byte*)offset)
+                    // This becomes a conditional jmp foward to not favor it.
+                    goto NotEqual;
+                }
+                // Use Vector128.Size as Vector128<byte>.Count doesn't inline at R2R time
+                // https://github.com/dotnet/runtime/issues/32714
+                else if (length >= Vector128.Size)
                 {
-                    if (LoadUIntPtr(ref first, offset) != LoadUIntPtr(ref second, offset))
+                    Vector128<byte> vecResult;
+                    nuint offset = 0;
+                    nuint lengthToExamine = length - Vector128.Size;
+                    if (lengthToExamine != 0)
                     {
-                        goto NotEqual;
+                        do
+                        {
+                            // We use instrincs directly as .Equals calls .AsByte() which doesn't inline at R2R time
+                            // https://github.com/dotnet/runtime/issues/32714
+                            vecResult = Sse2.CompareEqual(LoadVector128(ref first, offset), LoadVector128(ref second, offset));
+                            if (Sse2.MoveMask(vecResult) != 0xFFFF)
+                            {
+                                goto NotEqual;
+                            }
+                            offset += Vector128.Size;
+                        } while (lengthToExamine > offset);
                     }
-                    offset += sizeof(UIntPtr);
-                }
-                return LoadUIntPtr(ref first, lengthToExamine) == LoadUIntPtr(ref second, lengthToExamine);
-            }
 
-            Debug.Assert((byte*)lengthToExamine < (byte*)sizeof(UIntPtr));
+                    // Do final compare as Vector128<byte>.Count from end rather than start
+                    Debug.Assert(lengthToExamine >= 0);
+                    vecResult = Sse2.CompareEqual(LoadVector128(ref first, lengthToExamine), LoadVector128(ref second, lengthToExamine));
+                    if (Sse2.MoveMask(vecResult) == 0xFFFF)
+                    {
+                        // C# compiler inverts this test, making the outer goto the conditional jmp.
+                        goto Equal;
+                    }
 
-            // On 32-bit, this will never be true since sizeof(UIntPtr) == 4
-#if TARGET_64BIT
-            if ((byte*)lengthToExamine >= (byte*)sizeof(int))
-            {
-                if (LoadInt(ref first, offset) != LoadInt(ref second, offset))
-                {
+                    // This becomes a conditional jmp foward to not favor it.
                     goto NotEqual;
                 }
-                offset += sizeof(int);
-                lengthToExamine -= sizeof(int);
             }
-#endif
-
-            if ((byte*)lengthToExamine >= (byte*)sizeof(short))
+            else if (Vector.IsHardwareAccelerated && (nint)length >= Vector<byte>.Count)
```

Comment:
All of these should be `nuint` instead of `nint`. It's possible on 32-bit platforms that a value greater than 2GB will be passed for the _length_ parameter.

Reply:
Done